.github/workflows/dependency-check.yml

```diff
@@ -0,0 +1,49 @@
+name: Dependency Check
+
+on:
+  pull_request:
+    paths:
+      - 'go.mod'
+      - 'go.sum'
+      - '.github/workflows/dependency-check.yml'
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-latest
+    name: Dependency Check
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - id: all
+        run: |
+          echo "dependencies<<EOF" >> $GITHUB_OUTPUT
+          # Extract all dependencies from go.mod (include indirect dependencies and comments)
```

Comment:
```suggestion
          # Extract all dependencies from go.mod (include indirect dependencies and comments).
          # `go list` isn't used because:
          # 1. it lists ALL the transitive dependencies, even those that are unused and don't make it to the go.mod file
          # 2. It doesn't extract the inline `dependency-check-ignore` comments.
```